lib/cloudant.js

```diff
@@ -108,7 +108,8 @@ Cloudant.prototype.connect = function(cb) {
  * @param {Object} doc The model document/data
  */
 Cloudant.prototype.toDB = function(modelName, modelObject, doc) {
-  var idValue = this.getIdValue(modelName, doc);
+  var idValue = this.getIdValue(modelName, doc) &&
+     this.getIdValue(modelName, doc).toString();
```

Comment:
Let's not do this; it requires two calls to the same function, which means you're actually doing all that work twice!

```
// We toString this value because IDs must be strings: <put link to Cloudant API doc here>
var idValue = this.getIdValue(modelName, doc);
if (!idValue) { // do error stuff here } else { idValue = idValue.toString(); }
// etc
```